examples/experimental/README.md

```diff
@@ -0,0 +1,70 @@
+RFC: Toward TF Encrypted 1.0
+==================================
+This is a living design doc for TF Encrypted's first big refactor.
+
+T.O.C.
+1. Problems with current implementation
+2. Overview
+3. Details
+4. Protocol Requirements
+
+
+### Problems with current implementation
+- Too much logic is wrapped up in the Pond protocol, and the file is too large
+- It is not clear which methods a new protocol needs to override to conform to the public TFE API (this API is largely implicitly defined and not stated explicitly anywhere)
+- BackingTensors being wrapped up in factories are impossible to test
+- Excessive code duplication: each operation is implemented at least once for each BackingTensor and each protocol
+
+
+
+### Overview
+The guiding philosophy of the refactor will be to replace TFE's major features with native TensorFlow abstractions that are covered by the TF Stability Guarantees for the lifetime of TF 2.0.  The goals of doing so include:
+- Upgrading TF Encrypted to TensorFlow 2.0
+- Leverage as much of the existing TensorFlow ecosystem with minimal rewriting and/or maintenance
+- Continue to rely on the stability guarantee of the TensorFlow Public API
+- Improve our own code modularity, extensibility, and sustainability and reduce code duplication
+
+Concretely, we will replace our own type system of custom tensors with a light set of hooks over existing tensor types in TensorFlow.  The current Protocol class would be replaced by custom subclasses of `tf.distribute.Strategy` and `tf.distribute.StrategyExtended` that define the necessary protocol operations and handle any distributed computation that's required by them.
```

Comment:
I still feel like we will fail checks inside of TF if we don't inherit from tf.distribute.StrategyExtended -- I'll admit that I don't have an example of how we might fail off the top of my head, but it seems quite possible.  I also don't understand what the downside of using inheritance would be since Strategy and StrategyExtended classes are very lightweight by themselves, and they're designed to be extended for doing distributed computations in different ways (e.g. tf mesh).